### PR TITLE
Feature/user playlist song creations

### PR DESCRIPTION
**Pull request by shellygamlielly**
Support creating a new user, finding an existing user, creating a new playlist, getting a playlist, add songs to a playlist.

### CODE REVIEW

File: src/user.service.ts [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-@Injectable()
-export class UserService {
-  constructor(@InjectModel(User.name) private userModel: Model<User>) {}
-
-  async getuserBySpotifyId(spotifyId): Promise<UserDto> {
-    const user = await this.userModel.findOne({ spotifyId }).exec();
-    if (!user) {
-      throw new NotFoundException(`Resource with ID ${spotifyId} not found`);
-    }
-    return user;
```
Comment (ramiloif) [inline src/user.service.ts:16]: Return userDto not the entire mongo user

File: src/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  async getUserBySpotifyId(spotifyId): Promise<UserDto> {
+    const user = await this.userModel.findOne({ spotifyId }).exec();
+    if (!user) {
+      throw new NotFoundException(
+        `User with spotifyId: ${spotifyId} was not found`,
+      );
+    }
+    return {
+      email: user.email,
```
Comment (ramiloif) [inline src/user/user.service.ts:19]: retur also userID

File: src/user/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      spotifyId: user.spotifyId,
+    };
+  }
+
+  async createUser(spotifyId: string, email: string): Promise<string> {
+    const user = await this.userModel.create({ spotifyId, email });
+    return user.id;
```
Comment (ramiloif) [inline src/user/user.service.ts]: Return USerDto

File: src/playlist/playlist.controller.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Body, Controller, Get, Param, Post } from '@nestjs/common';
+import { PlaylistService } from './playlist.service';
+import { PlaylistDto } from '../dto/playlist-dto';
+
+@Controller('/playlist')
+export class PlaylistController {
+  constructor(private readonly playlistService: PlaylistService) {}
+
+  @Get('/:ownerId')
```
Comment (ramiloif) [inline src/playlist/playlist.controller.ts]: add owner id here ( dont forget the client!!!!)

File: src/playlist/playlist.service.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    return playlists.map((playlist) => ({
+      ownerId: playlist.ownerId,
+      playlistId: playlist.id,
+      name: playlist.name,
+      maxTime: playlist.maxTimeSeconds,
+      songsCount: songsCountMap[playlist.id] || 0, // Default to 0 if not found
+    }));
+  }
+
+  async getPlaylistsByUserId(userId: ObjectId) {
```
Comment (ramiloif) [inline src/playlist/playlist.service.ts:56]: get all user related playlists

File: src/playlist/playlist.service.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    //any permission
+  }
+
+  async findPlaylistIdByName(
```
Comment (ramiloif) [inline src/playlist/playlist.service.ts]: What this neded for?

File: src/song/song.controller.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  @Post('')
+  async addSongToPlaylist(@Body() songDto: SongDto): Promise<ObjectId> {
+    const songId = await this.songService.addSongToPlaylist(
+      songDto.playlistId,
+      songDto.spotifySongId,
+    );
+    return songId;
+  }
+
+  @Post('/vote/:userId')
```
Comment (ramiloif) [inline src/song/song.controller.ts]: PUT

File: src/song/song.controller.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Delete('/:playlistId/:spotifySongId')
+  async removeSongFromPlaylist(
+    @Param('playlistId') playlistId: string,
+    @Param('spotifySongId') spotifySongId: string,
+  ): Promise<void> {
+    await this.songService.removeSongFromPlaylist(playlistId, spotifySongId);
+  }
+
+  @Delete('/Unvote/:userId')
```
Comment (ramiloif) [inline src/song/song.controller.ts]: lower case

File: src/song/song.controller.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  async removeVoteToSong(
+    @Body() songDto: SongDto,
+    @Param('userId') userId: string,
+  ): Promise<number> {
+    return await this.songService.removeVoteToSong(
```
Comment (ramiloif) [inline src/song/song.controller.ts:54]: unvoteSong

File: src/song/song.service.ts [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+    spotifySongId: string,
+    userId: string,
+  ): Promise<number> {
+    const updatedSong = await this.songModel
+      .findOneAndUpdate(
+        { playlistId, spotifySongId },
+        {
+          $addToSet: { voters: userId },
+        },
+        { new: true },
```
Comment (ramiloif) [inline src/song/song.service.ts:105]: ensure new: true not create a new song if not exist

File: src/playlist/playlist.service.ts (hunk 5 above)
Comment (shellygamlielly) [reply src/playlist/playlist.service.ts:56]: added getPlaylistCollaborators(playlistId: ObjectId)

**Pull request merged**